web/src/components/openai/constants.ts

```diff
@@ -1 +1 @@
-export const OPENAI_API_KEY_URL = "/api/manage/admin/openai-api-key";
```

Comment:
These files are still tied to OpenAI, I think they could be renamed to GenerativeAI or whatever right? Thoughts?

Reply:
Seems fine to me (although we may still need slightly different forms based on the service provider, not 100% sure). For example, we may want to allow people to select their model via the UI and the model choices would be different for each provider. Additionally, some providers may require a URL to be specified.

Reply:
Ya, one way this could be done is we have a mapping of valid settings/models internally but allow them to provide freeform values with the warning that it may not work.

But this doesn't really relate to this particular change. I'm just asking if you'd be down to change the frontend code to remove the hard assumption on OpenAI. I can do it but it's much safer/easier if you do it, probably takes 10 minutes tops